tests/lazy_fixture.py

```diff
@@ -0,0 +1,89 @@
+import dataclasses
+import typing
+
+import pytest
+
+
+@dataclasses.dataclass
+class LazyFixture:
+    """Lazy fixture dataclass."""
+
+    name: str
+
+
+def lazy_fixture(name: str) -> LazyFixture:
+    """Mark a fixture as lazy."""
+    return LazyFixture(name)
+
+
+def is_lazy_fixture(value: object) -> bool:
+    """Check whether a value is a lazy fixture."""
+    return isinstance(value, LazyFixture)
+
+
+def pytest_make_parametrize_id(
+    config: pytest.Config,
+    val: object,
+    argname: str,
+) -> typing.Optional[str]:
+    """Inject lazy fixture parametrized id.
+
+    Reference:
+    - https://bit.ly/48Off6r
+
+    Args:
+        config (pytest.Config): pytest configuration.
+        value (object): fixture value.
+        argname (str): automatic parameter name.
+
+    Returns:
+        typing.Optional[str]: new parameter id.
+    """
+    if is_lazy_fixture(val):
+        return typing.cast(LazyFixture, val).name
+    return None
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_fixture_setup(
+    fixturedef: pytest.FixtureDef,
+    request: pytest.FixtureRequest,
+) -> typing.Optional[object]:
+    """Lazy fixture setup hook.
+
+    This hook will never take over a fixture setup but just simply will
+    try to resolve recursively any lazy fixture found in request.param.
+
+    Reference:
+    - https://bit.ly/3SyvsXJ
```

Comment:
WDYT about having a non-reduced URL here, to be able to find what it was about again if the link is ever broken?